### ISSUE TITLE
WRONLY should not be corrected

### ISSUE THREAD

**Issue by SOF3** [crate-ci/typos#744]
`O_WRONLY` should be ignored by default as it is a standard constant in libc.

**Comment by SOF3** [comment]
I checked the C and Go libraries, they seem to be using `O_WRONLY` (`os.O_WRONLY`) as a whole instead of just `WRONLY`. Not sure about other languages.

**Comment by groengpx** [comment]
Yep, I got the same issue since the last update. I would indeed say that `O_WRONLY` should be marked a whole as a valid identifier. I personally recommend against allowing `WRONLY`, since that seems a typical spelling error to me.

**Comment by groengpx** [comment]
Can I contribute this fix myself? If so, is https://github.com/crate-ci/typos/blob/84c8b30e065012c70187818aadba0e711ff5cafc/crates/typos-dict/assets/allowed.csv the correct file to edit for this case?

**Comment by epage** [comment]
`allowed.csv` is a way for us to remember what words we intentionally removed from `words.csv`.  In this case, we don't want to mark `wronly`  as an allowed word, removing it from `words.csv` as that is broader than what is needed.

Instead we should be marking `O_WRONLY` as a valid identifier which will cause it to never be split into words to then be individually correct.  `O_WRONLY` likely appears in enough languages that bind to libc that we don't want to just add a C-specific list of valid identifiers like we have for some other languages.  However, we currently don't have a list of global identifier corrections.  We might be able to fake it in the config if all of the layering is done correctly.

**Comment by groengpx** [comment]
The fix should be in 1.16.14, right? I've just updated to that version, but it still tries to correct `O_WRONLY`.

```
$ typos --version
typos-cli 1.16.14
                                                                                                                                                                    (0)
$ typos
error: `WRONLY` should be `WRONGLY`
  --> ./redacted.c:15:38
   |
15 |   int fd = open(path, O_NONBLOCK | O_WRONLY);
   |                                      ^^^^^^
   |
```

**Comment by epage** [comment]
Looks like we had a bug where were only checked the built-in idents dictionary if the user also had idents configured.  A new release will be out soon

**Comment by groengpx** [comment]
Confirmed fixed with commit 243b1879346b307640e040472e199c52ada54bf9 (in release 1.16.15)

Thanks!
